phastSim/phastSim.py

```diff
@@ -607,14 +693,96 @@ def normalize(node, norm):
                 node.rate /= norm
                 if node.isTerminal:
                     node.rates /= norm
+
                 else:
                     normalize(node.belowNodes[0], norm)
                     normalize(node.belowNodes[1], norm)
 
         # normalize the rates
         normalize(node=self.genomeRoot, norm=norm)
 
+        if self.indels:
+            # if there are indels then some further calculation is require to convert the rates from 
+            # substitution rates to overall rates.
+            
+
+            def include_indels(node, scale):
+                
+                if node.isTerminal:
+                    node.insertionRate * scale
+                    node.deletionRate * scale
+                    node.rate += (node.insertionRate + node.deletionRate)
+                
+                else:
+                    node.rate = include_indels(node.belowNodes[0], scale) + include_indels(node.belowNodes[1], scale)
+
+                return node.rate
+
+            include_indels(node=self.genomeRoot, scale=scale)
```

Comment:
I don't think it's wise to call include_indels() before normalizeRates().
The reason is that when we normalize the rates, the total rate self.genomeRoot.rate is supposed to be the total substitution rate, not the total mutation rate, but when you call include_indels() on the root then self.genomeRoot.rate becomes the total mutation rate.

Reply:
Actually, what is the purpose of calling normalizeRates() below? At the end of this function? Here the substitution rates where already normalized, and the indel rates have already been scaled and added, so I suspect that self.normalizeRates() at line 729 below should not be there.

Reply:
See below - most of this function is just commented out (for example in my line 722 is the closing """ comment)